### ISSUE TITLE
ldjam.com SSL error

### ISSUE THREAD

**Issue by local-minimum** [JammerCore/JammerCore#2150]
The site is not reachable, the request returns the error code: SSL_ERROR_NO_CYPHER_OVERLAP

**Comment by mikekasprzak** [comment]
What browser/OS?

**Comment by mikekasprzak** [comment]
Google says that's a Firefox error, and offers some unusual user-side fixes.

To make sure, I ran some tests, and current SSL testing practices say we are doing everything right.

https://www.ssllabs.com/ssltest/analyze.html?d=ldjam.com&s=23.205.154.163

Notably we don't have TLS 1.1 enabled anymore, since that's considered insecure.

**Comment by local-minimum** [comment]
![20220910_180006](https://user-images.githubusercontent.com/8041100/189491713-09e3e3ac-6cd4-4fde-99b8-f9d137a872c7.jpg)
On android 10 if I was on mobile.  But today I get a cloudflair dns error

**Comment by mikekasprzak** [comment]
That's odd. We're not on CloudFlare anymore.

**Comment by mikekasprzak** [comment]
Just to confirm, are we talking about `ldjam.com` or `ludumdare.com`?

**Comment by local-minimum** [comment]
ldjam.com 
![Screenshot_20220910-180536_Firefox](https://user-images.githubusercontent.com/8041100/189491874-c11645d9-bc0c-4334-b103-219749c8d4bf.jpg)

**Comment by local-minimum** [comment]
ldjam.com 
![Screenshot_20220910-180536_Firefox](https://user-images.githubusercontent.com/8041100/189491874-c11645d9-bc0c-4334-b103-219749c8d4bf.jpg)

**Comment by mikekasprzak** [comment]
Okay, so for whatever reason you're DNS hasn't been updated. This was changed a few weeks ago.

Was:

```
ldjam.com.		86400	IN	NS	angela.ns.cloudflare.com.
ldjam.com.		86400	IN	NS	ned.ns.cloudflare.com.

```

Now:
```
ldjam.com.		86400	IN	NS	a12-64.akam.net.
ldjam.com.		86400	IN	NS	a9-65.akam.net.
ldjam.com.		86400	IN	NS	a13-66.akam.net.
ldjam.com.		86400	IN	NS	a20-65.akam.net.
ldjam.com.		86400	IN	NS	a1-172.akam.net.
ldjam.com.		86400	IN	NS	a24-65.akam.net.
```

**Comment by local-minimum** [comment]
Alright,  strange,  but not really a ldjam.com issue then.

**Comment by mikekasprzak** [comment]
Please let me know if you manage to resolve this. Also let me know if you're using a non-standard DNS server (1.1.1.1, 8.8.8.8, 9.9.9.9, etc) or just whatever your ISP assigns you.

Chances are I might get someone else running into the same issues as you, so any details you can provide on how you resolved it would greatly help. Thanks!

**Comment by local-minimum** [comment]
No dns settings on my phone connecting over wifi so probably the isp? Or something my old Samsung thinks is default.

**Comment by local-minimum** [comment]
Using my mobile isp or vpn resolves it so it seems my broadband isp has some issues

**Comment by local-minimum** [comment]
Actually I got annoyed and attempted on my windows computer to set DNS to 1.1.1.1 and I still can't connect. I attempted some power shell, but to be honest this really isn't my domain. The answers from lookups were:

```powershell
PS > Resolve-DnsName -Name ldjam.com -Type A -DnsOnly
ldjam.com                                      A      29703 Answer     188.114.97.2
ldjam.com                                      A      29703 Answer     188.114.96.2

PS > Resolve-DnsName -Name ldjam.com -Type A -Server 1.1.1.1
ldjam.com                                      A      20    Answer     184.51.252.138
ldjam.com                                      A      20    Answer     184.51.252.162

PS > Resolve-DnsName -Name ldjam.com -Type A -Server 1.0.0.1
ldjam.com                                      A      20    Answer     184.51.252.138
ldjam.com                                      A      20    Answer     184.51.252.162

PS > Resolve-DnsName -Name ldjam.com -Type A -Server 8.8.8.8 
ldjam.com                                      A      20    Answer     23.197.207.234
ldjam.com                                      A      20    Answer     23.197.207.139
```

and whichever I try, none of these ip seems to lead to ldjam.com if I enter them in the browser but to an `Error: The requested URL "[no URL]", is invalid`.

With DNS set to 1.1.1.1 and having flushed my dns cache both in firefox and on windows I still get the cipher error going to ldjam.com....

**Comment by mikekasprzak** [comment]
You might be on to something, but I'm not sure what, so lets start by ruling out DNS.

For reference, the whois record for `ldjam.com` looks like this.

```
mike@thelastx270:~$ whois ldjam.com

   Domain Name: LDJAM.COM
   Registry Domain ID: 1872453402_DOMAIN_COM-VRSN
   Registrar WHOIS Server: whois.porkbun.com
   Registrar URL: http://porkbun.com
   Updated Date: 2022-08-29T07:24:40Z
   Creation Date: 2014-08-23T19:53:56Z
   Registry Expiry Date: 2023-08-23T19:53:56Z
   Registrar: Porkbun LLC
   Registrar IANA ID: 1861
   Registrar Abuse Contact Email: abuse@porkbun.com
   Registrar Abuse Contact Phone: 5038508351
   Domain Status: clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited
   Domain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited
   Name Server: A1-172.AKAM.NET
   Name Server: A12-64.AKAM.NET
   Name Server: A13-66.AKAM.NET
   Name Server: A20-65.AKAM.NET
   Name Server: A24-65.AKAM.NET
   Name Server: A9-65.AKAM.NET
   DNSSEC: signedDelegation
   DNSSEC DS Data: 2779 13 2 EE3A962044BB26E2D3FDD378A515C58954943B888EA2126FC8B4E9001AC75A04
   URL of the ICANN Whois Inaccuracy Complaint Form: https://www.icann.org/wicf/
>>> Last update of whois database: 2022-09-14T17:14:44Z <<<
```
Notably the name servers in the `whois` point at Akamai (`AKAM.NET`).

When I dig for `a` records, I get the following:

```bash
# Local DNS
mike@thelastx270:~$ dig a ldjam.com +short
206.248.168.177
206.248.168.179
# Akamai DNS
mike@thelastx270:~$ dig a ldjam.com +short @a1-172.akam.net
206.248.168.177
206.248.168.179
# Google DNS
mike@thelastx270:~$ dig a ldjam.com +short @8.8.8.8
206.248.168.177
206.248.168.179
# CloudFlare DNS
mike@thelastx270:~$ dig a ldjam.com +short @1.1.1.1
23.61.246.32
23.61.246.34
# Quad9 DNS
mike@thelastx270:~$ dig a ldjam.com +short @9.9.9.9
23.61.246.163
23.61.246.154
```

The specific IP's aren't important, as when you reverse proxy a website through a CDN, different IPs will be returned depending on where you are in the world. In my tests above, myself, Akamai, and Google all return the same IPs, but CloudFlare and Quad9 do not.

That said, that doesn't necessarily mean anything is wrong.

```bash
# Local DNS
mike@thelastx270:~$ dig ns ldjam.com +short
a12-64.akam.net.
a24-65.akam.net.
a1-172.akam.net.
a9-65.akam.net.
a13-66.akam.net.
a20-65.akam.net.
# Akamai DNS
mike@thelastx270:~$ dig ns ldjam.com +short @a1-172.akam.net
a12-64.akam.net.
a1-172.akam.net.
a20-65.akam.net.
a9-65.akam.net.
a24-65.akam.net.
a13-66.akam.net.
# Google DNS
mike@thelastx270:~$ dig ns ldjam.com +short @8.8.8.8
a24-65.akam.net.
a1-172.akam.net.
a13-66.akam.net.
a9-65.akam.net.
a20-65.akam.net.
a12-64.akam.net.
# CloudFlare DNS
mike@thelastx270:~$ dig ns ldjam.com +short @1.1.1.1
a9-65.akam.net.
a1-172.akam.net.
a12-64.akam.net.
a13-66.akam.net.
a20-65.akam.net.
a24-65.akam.net.
# Quad9 DNS
mike@thelastx270:~$ dig ns ldjam.com +short @9.9.9.9
a12-64.akam.net.
a9-65.akam.net.
a1-172.akam.net.
a20-65.akam.net.
a13-66.akam.net.
a24-65.akam.net.
```

Which shows that all the IPs I get are using Akamai's nameservers.

So, can I get you to run `Resolve-DnsName -Name ldjam.com -Type NS -DnsOnly` to check the nameservers? (I'm assuming `-Type NS` will tell you that).

**Comment by mikekasprzak** [comment]
Ah sorry, `Resolve-DnsName` is equivalent to `nslookup` not dig. My results from that are as follows.

```bash
mike@thelastx270:~$ nslookup ldjam.com
Server:		127.0.0.53
Address:	127.0.0.53#53

Non-authoritative answer:
Name:	ldjam.com
Address: 206.248.168.177
Name:	ldjam.com
Address: 206.248.168.179
Name:	ldjam.com
Address: 2600:140a::172b:f288
Name:	ldjam.com
Address: 2600:140a::172b:f289

mike@thelastx270:~$ nslookup ldjam.com a1-172.akam.net
Server:		a1-172.akam.net
Address:	193.108.91.172#53

Name:	ldjam.com
Address: 206.248.168.179
Name:	ldjam.com
Address: 206.248.168.177
Name:	ldjam.com
Address: 2607:f2c0:1:3303::cef8:a8b1
Name:	ldjam.com
Address: 2607:f2c0:1:3303::cef8:a8b3

mike@thelastx270:~$ nslookup ldjam.com 8.8.8.8
Server:		8.8.8.8
Address:	8.8.8.8#53

Non-authoritative answer:
Name:	ldjam.com
Address: 206.248.168.179
Name:	ldjam.com
Address: 206.248.168.177
Name:	ldjam.com
Address: 2607:f2c0:1:3303::cef8:a8b3
Name:	ldjam.com
Address: 2607:f2c0:1:3303::cef8:a8b1

mike@thelastx270:~$ nslookup ldjam.com 1.1.1.1
Server:		1.1.1.1
Address:	1.1.1.1#53

Non-authoritative answer:
Name:	ldjam.com
Address: 23.61.246.32
Name:	ldjam.com
Address: 23.61.246.34
Name:	ldjam.com
Address: 2600:140a:e000::173d:f620
Name:	ldjam.com
Address: 2600:140a:e000::173d:f622

mike@thelastx270:~$ nslookup ldjam.com 9.9.9.9
Server:		9.9.9.9
Address:	9.9.9.9#53

Non-authoritative answer:
Name:	ldjam.com
Address: 23.61.246.163
Name:	ldjam.com
Address: 23.61.246.154
Name:	ldjam.com
Address: 2600:140a:e000::173d:f6a3
Name:	ldjam.com
Address: 2600:140a:e000::173d:f69a
```

If I was to change my `hosts` file, `dig` wouldn't report that, but nslookup would.

```bash
mike@thelastx270:~$ nslookup ldjam.com
Server:		127.0.0.53
Address:	127.0.0.53#53

Name:	ldjam.com
Address: 1.2.3.4
```

**Comment by mikekasprzak** [comment]
Something else to check for, Firefox may have DNS over HTTPS enabled. This isn't necessarily a problem, but if we're getting inconsistent results between the terminal and Firefox, this could be why.

![dnsshield](https://user-images.githubusercontent.com/1570313/190231642-dc38b44d-89fc-44cf-92e1-442d3d20bfea.png)

![dnsshield2](https://user-images.githubusercontent.com/1570313/190231696-9783cf6d-e0ae-4991-9062-f31b9d3c8f53.png)

Months ago, my Firefox had CIRA Canadian Shield set (Default) without me realizing it. Funny thing, Canadian Shield is a partnership between the Canadian government (CIRA) and Akamai. I'm guessing if you're outside Canada then CloudFlare is used instead.

**Comment by local-minimum** [comment]
So all the lookups with `-Type NS` gets `*.akam.net` and I believe when I looked up the error messages I got from using just the IP they were all pointing to that place too, which seems about right.

Let me check the other settings... but it's still weird that it seems to go away when I'm using my mobile carriers internet connection on my phone or go through a VPN...

**Comment by mikekasprzak** [comment]
Okay, I just realized I could run some reverse IP lookups on those returned IPs you provided, see who owns the IP blocks.

* `188.114.97.2` - CLOUDFLARENET
* `184.51.252.138` - AKAMAI INTERNATIONAL BV
* `23.197.207.234` - AKAMAI INTERNATIONAL BV

So yes, it looks like `Resolve-DnsName -Name ldjam.com -Type A -DnsOnly` returned an IP belonging to CloudFlare, but the lookups where you specified the servers (including CloudFlare's server 1.1.1.1) correctly returned Akamai.

**Comment by mikekasprzak** [comment]
> Alright if I activate DNS over HTTPS and choose NextDNS I can reach ldjam.com, but if I choose Cloudflare I cannot....

Okay interesting! I wonder if some of the CloudFlare DNS servers prefer their internal data, despite what the `whois` says. I technically still have a configuration on CloudFlare for `ldjam.com`. I might need to delete it for those servers to give it up.

**Comment by local-minimum** [comment]
And I just realized I have two internet connections and when I override both to use 1.1.1.1 then everything is gucci... so it does seem like I need to have a talk with my broadband ISP about what they are up to with their DNS.... 

It's a bit strange that the CloudFlare DNS over HTTPS so might be a good thing to delete that config yes.

**Comment by mikekasprzak** [comment]
Okay great, I'm glad we found the cause. 

> I technically still have a configuration on CloudFlare for ldjam.com. I might need to delete it for those servers to give it up.

Nevermind, I have already deleted my `ldjam.com` CloudFlare configuration.

**Comment by mikekasprzak** [comment]
My apologies on the chat situation. In the mean time yes, here on GitHub is best. There's a new Discussions section for tangents. ;)

The Gitter got left in a confusing state after GitLab acquired them, and I wasn't sure how to proceed. Not to mention I did rename the GitHub repositories. I was also in the process of setting up a Matrix server, and I believe I got a bridge to the Gitter working, but yeah the room was dead so finishing that up fell below other priorities.

> the spam comment situation, do you have a roadmap

Yes we're doing a few things. Akamai is helping me collect data on suspicious requests we're getting, but that's only been live for a week. Early results said 69% of all traffic was attempted attacks (such things as requests for URLs only found on WordPress websites). So on that front we're data gathering.

There's a big upgrade I've been finishing up that fixes a number of core and UX problems with the website (such as navigating or using the back button on posts). Features like "flagging users" are sort of dependent on that, since I'd rather add that to the fixed code than the current live code. This upgrade was pushed back a few times as I finished client work and I fixed bugs (game listing is broken as I type this). That said, I quite literally today got approval from the client, so I'm finally able to focus exclusively on the upgrade code. In the next couple days I'll have hopefully finished and published the upgrade branch, which should make fixing UX problems a lot easier, we can finally add a UX for changing usernames and post/game slugs, add actual support for deleting content (:sweat_smile:), and some tools for dealing with the spammers.

TL;DR: soon.

**Comment by local-minimum** [comment]
Great entry on ldjam!

And I  think I found the problem. I contacted my ISP and they didn't have any issues with ldjam.com then I  dug through my asus router settings and discovered that it's dns was set to default.  Whatever that was,  by changing it to custom setting quad 1 it solved the problem on all my devices.

**Comment by mikekasprzak** [comment]
Great, I'm glad you found the culprit.
